api/src/main/java/org/infinispan/api/common/events/KeyValueListener.java

```diff
@@ -0,0 +1,53 @@
+package org.infinispan.api.common.events;
+
+/**
+ * KeyValueListener interface to be used with caches
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 10.0
+ */
+public interface KeyValueListener<K, V> {
+
+   default boolean sync() {
```

Comment:
The more I think about it, I don't know if we need this any longer. This should be able to be handled by the user. Since the method can determine if it is non blocking or not the user can return null right away and process async themselves.

Reply:
Only if you guys explicitly want to eliminate your listener thread pool - otherwise this is quite convenient for users (myself included) - who would otherwise need to maintain their own thread pool so not to block the listener thread.  I'm fine if you want to remove it, but I just wanted to acknowledge its utility.

Reply:
I'm not convinced, I don't think it ever was the best place to run blocking user code: it has only 1 thread by default, in order to preserve the ordering of notifications, and if one thread can't keep up you have to increase the pool size and lose ordering for all caches.

@pferraro I thought you had plenty of thread pools to choose from in WildFly :)

Reply:
We do :) but other users might not.